lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -3049,15 +3049,18 @@ def initialize(self, connection):
         super().initialize(connection)
 
         # https://www.postgresql.org/docs/9.3/static/release-9-2.html#AEN116689
-        self.supports_smallserial = self.server_version_info >= (9, 2)
+        self.supports_smallserial = self.server_version_info and self.server_version_info >= (9, 2)
 
-        self._set_backslash_escapes(connection)
+        try: 
+            self._set_backslash_escapes(connection)
+        except:
+            pass
 
-        self._supports_drop_index_concurrently = self.server_version_info >= (
+        self._supports_drop_index_concurrently = self.server_version_info and self.server_version_info >= (
             9,
             2,
         )
-        self.supports_identity_columns = self.server_version_info >= (10,)
+        self.supports_identity_columns = self.server_version_info and self.server_version_info >= (10,)
```

Comment:
server_version_info must be provided here, it's invalid state for this to not be present